### PR TITLE
fix: improve docs on how to define an extension

### PR DESCRIPTION
**Pull request by ruheni**
The PR improves the docs on how to define and use extensions

### CODE REVIEW

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx [hunk 1]
```diff
@@ -63,26 +63,47 @@ generator client {
...
 
-```ts
-const xprisma = prisma.$extends({
-  name: `signUp`, // Optional: name appears in error logs
-  model: {        // This is a `model` component
-    user: { ... } // The extension logic for the `user` model goes inside the curly braces
-  },
-})
-```
+   ```ts
```
Comment (jharrell) [inline content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx:70]: should this be indented by 3 spaces? just double checking.

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx [hunk 2]
```diff
@@ -63,26 +63,47 @@ generator client {
...
+   import { Prisma } from '@prisma/client'
+
+   // Define the extension
+   const myExtension = Prisma.defineExtension({
+       name: 'signUp', // Optional: name appears in error logs
+       model: {        // This is a `model` component
+       user: { ... } // The extension logic for the `user` model goes inside the curly braces
+     },
+   })
```
Comment (jharrell) [inline content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx]:
```suggestion
  ```ts
  import { Prisma } from '@prisma/client'

  // Define the extension
  const myExtension = Prisma.defineExtension({
    name: 'signUp', // Optional: name appears in error logs
    model: {        // This is a `model` component
      user: { ... } // The extension logic for the `user` model goes inside the curly braces
    },
  })
```

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx [hunk 3]
```diff
@@ -63,26 +63,47 @@ generator client {
...
-})
-```
+   ```ts
+   const xprisma = prisma.$extends({
+    name: 'signUp', // Optional: name appears in error logs
+    model: {        // This is a `model` component
+      user: { ... } // The extension logic for the `user` model goes inside the curly braces
+    },
+   })
+   ```
```
Comment (jharrell) [inline content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx]:
```suggestion
  ```ts
  const xprisma = prisma.$extends({
    name: 'signUp', // Optional: name appears in error logs
    model: {        // This is a `model` component
      user: { ... } // The extension logic for the `user` model goes inside the curly braces
    },
  })
  ```
```

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx (hunk 1 above)
Comment (ruheni) [reply content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx:70]: I added 4 spaces and the precommit hook removed one space 😅

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx [hunk 4]
```diff
@@ -63,26 +63,47 @@ generator client {
...
+
+2. Use the `Prisma.defineExtension` method and pass it to the `$extends` client-level method
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx]: what is "it" that you parse? You don't pass the method (which the "it" currently would naturally resolve to)

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx [hunk 5]
```diff
@@ -63,26 +63,47 @@ generator client {
...
+
+   ```ts
+   import { Prisma } from '@prisma/client'
+
+   // Define the extension
+   const myExtension = Prisma.defineExtension({
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx:85]: Is this a normal, default way to define an extension?
Isn't this more for creating a re-usable extension that is packaged up? If so, shouldn't this be mentioned here (at least via dedicated documentation about that)?

Comment (ruheni) [reply content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx:85]: This is normal for users who want to define multiple extensions in a separate directory within a project without packaging it up into a separate package(s).

See user comment in discussion: https://github.com/prisma/prisma/discussions/19725#:~:text=I%20would%20like%20to%20organise%20my%20client%20extensions%20in%20a%20separate%20dir%2C%20for%20the%20sake%20of%20maintainability%3A

Comment (janpio) [reply content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx:85]: Then let's give such context so you can understand the use cases.

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx (hunk 4 above)
Comment (ruheni) [reply content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx]: I've updated the statement. Let me know if it reads better

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx [hunk 6]
```diff
@@ -63,26 +63,53 @@ generator client {
 
 ## Create an extension
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx]:
```suggestion
## Add an extension to Prisma Client
```

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx [hunk 7]
```diff
@@ -63,26 +63,53 @@ generator client {
...
 
-Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an extended client. In your `$extends` structure, use the appropriate extension component or components ([`model`](/concepts/components/prisma-client/client-extensions/model), [`client`](/concepts/components/prisma-client/client-extensions/client), [`result`](/concepts/components/prisma-client/client-extensions/result) or [`query`](/concepts/components/prisma-client/client-extensions/query)).
+You can create an extension using two primary ways. You can either:
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx]:
```suggestion
You can create an extension using two primary ways:
```

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx [hunk 8]
```diff
@@ -61,28 +61,55 @@ generator client {
...
 ```
 
-## Create an extension
+## Add an extension to Prisma Client
 
-Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an extended client. In your `$extends` structure, use the appropriate extension component or components ([`model`](/concepts/components/prisma-client/client-extensions/model), [`client`](/concepts/components/prisma-client/client-extensions/client), [`result`](/concepts/components/prisma-client/client-extensions/result) or [`query`](/concepts/components/prisma-client/client-extensions/query)).
+You can create an extension using two primary ways. You can either:
 
-For example, the following code uses a [`model` extension component](/concepts/components/prisma-client/client-extensions/model) to extend the `User` model.
+1. Use `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods)
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx]:
```suggestion
- Use the client-level [`$extends`](/reference/api-reference/prisma-client-reference#client-methods) method
```

File: content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx [hunk 9]
```diff
@@ -61,28 +61,55 @@ generator client {
...
+   ```ts
+   const xprisma = prisma.$extends({
+     name: 'signUp', // Optional: name appears in error logs
+     model: {        // This is a `model` component
+       user: { ... } // The extension logic for the `user` model goes inside the curly braces
+     },
+   })
+   ```
+
+2. Use the `Prisma.defineExtension` method to define an extension and assign it to a variable, and then pass the extension to the `$extends` client-level method
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx]:
```suggestion
- Use the `Prisma.defineExtension` method to define an extension and assign it to a variable, and then pass the extension to the client-level `$extends`  method
```

**Comment by nikolasburk** [comment]
This PR has a fairly simple change targeting only one section but the merge conflict was difficult to resolve. I re-created the changes in another PR and merged it, so closing this one: https://github.com/prisma/docs/pull/5499

**Pull request closed without merging**